src/core/parser.js

```diff
@@ -507,8 +507,12 @@ var Parser = (function ParserClosure() {
             stream.pos += i - ENDSTREAM_SIGNATURE_LENGTH;
             break;
           }
-          skipped += scanLength;
-          stream.pos += scanLength;
+          // Since the stream is scanned in `SCAN_BLOCK_SIZE` sized chunks,
+          // it's thus possible that an 'endstream' might be split between
+          // two contiguous `scanBytes` chunks. Avoid skipping over a valid
+          // 'endstream' by subtracting `j` here (fixes issue1536.pdf).
+          skipped += scanLength - j;
+          stream.pos += scanLength - j;
```

Comment:
Oh, I see it's just bad loop structure above. We shall re-write it as:

  while(i < scanLength) {
    j = 0;
    while (j < ENDSTREAM_SIGNATURE_LENGTH && scanBytes[i + j] === ENDSTREAM_SIGNATURE[j]) j++;
    if (j >= ENDSTREAM_SIGNATURE_LENGTH) {
                 found = true;
                 break;
    }
    i++;
  }

which easier-to-read equivalent of the code above without error.